src/metrics.py

```diff
@@ -0,0 +1,43 @@
+"""Handle submitting metrics, logs and other interesting details about jobs."""
+
+from os import getenv as env
+from typing import Any
+
+from prometheus_client import CollectorRegistry, Gauge, push_to_gateway
+
+
+def log_job_metrics(job_metrics: dict[str, Any]) -> None:
+    """Log metrics about a job to a prometheus pushgateway."""
+    if not (prometheus_pushgateway_url := env("PROMETHEUS_PUSHGATEWAY_URL")):
+        return
+
+    registry = CollectorRegistry()
+
+    job_success_timestamp = Gauge(
+        name="job_last_success_unixtime",
+        documentation="Unix timestamp of job end",
+        registry=registry,
+    )
+    job_success_timestamp.set_to_current_time()
+
+    # this is intentionally a Gauge rather than a Counter as per the prometheus
+    # documentation: "Do not use a counter to expose a value that can decrease."
```

Comment:
But can it decrease? if it succeeds the count is 1 if it fails the count is 0. On each successive run, the count either increases or remains the same...

Reply:
That's a good point, I was thinking along the lines of - users can set the counter to zero via API calls to the monitoring system whenever they fix an issue with a failing job. But you might be right, I'll switch it to a counter and we can roll back to a gauge later if it turns out it's doing the wrong thing.